Resolve regression to to use clause

montegoulding opened the pull request:
This commit resolves a regression where a use clause is followed by an
on .. call ... clause the parser will look ahead to the next line
and treat the on as part of the use definition. The option to ensure
the matched token is in the same row as the parsers current position
has been added to all the `ParserWillMatch...` functions. There are very
likely other functions in the parser that could make use of the extra
option.

peter-b commented:
@montegoulding I'm 99.9% certain that you don't mean to add a YAML library in this pull request.

@livecode-vulcan review not ok 6100462

livecode-vulcan commented:
💔 review by @peter-b not ok 61004625ea302d102d682e02b9191a8b299282e1

montegoulding commented:
make that 100%

Commit:
Resolve regression to use clause

This commit resolves a regression where a use clause is followed by an
on .. call ... clause the parser will look ahead to the next line
and treat the on as part of the use definition. The option to ensure
the matched token is in the same row as the parsers current position
has been added to all the ParserWillMatch... functions. There are very
likely other functions in the parser that could make use of the extra
option.

montegoulding commented:
@peter-b sorry about that

peter-b commented:
@livecode-vulcan review ok 47c8961

livecode-vulcan commented:
💙 review by @peter-b ok 47c89617fdb326c22c8428e9508291bf9ef94d54

livecode-vulcan commented:
😎 test success 47c89617fdb326c22c8428e9508291bf9ef94d54
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/1224)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/1236)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/1221)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/1229)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/1240)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/1240)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/1244)
